tests/containers/podman_integration.pm

```diff
@@ -0,0 +1,83 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: podman integration
+# Summary: Upstream podman integration tests
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use serial_terminal qw(select_serial_terminal select_user_serial_terminal);
+use containers::utils qw(get_podman_version);
+use utils qw(zypper_call script_retry);
+use version_utils qw(get_os_release);
+use containers::common;
+
+sub run {
+    select_serial_terminal;
+    my ($running_version, $sp, $host_distri) = get_os_release;
+    install_podman_when_needed($host_distri);
+
+    # Install tests dependencies
+    my @pkgs = qw(apache2-utils bats buildah criu go gpg2 jq make netcat-openbsd openssl podman-remote python3-PyYAML skopeo socat sudo systemd-container);
+    zypper_call "in @pkgs";
+
+    my $podman_version = get_podman_version();
+    script_retry("curl -sL https://github.com/containers/podman/archive/refs/tags/v$podman_version.tar.gz | tar -zxf -", retry => 5, delay => 60, timeout => 300);
+    assert_script_run "cd podman-$podman_version/";
+    # The user must be able to create a log file too
```

Comment:
Just do it in the user home instead?

Reply:
If you put the podman code into the user home, root can access it as well without any permission changes

Reply:
Yes, but do we have a way to know which user is configured in each release?

Reply:
openQA set up the OS, so it's consistent. You can use e.g. `assert_script_run "cd ~$testapi::username";` as root and just the CWD as user. Or use `/opt` or some other location where users have at least read privs for the podman code, but ideally not `/tmp` to avoid bloating RAM use. The log files can be anywhere IMO